bindings/c/src/lib.rs

```diff
@@ -152,6 +152,40 @@ pub unsafe extern "C" fn libsql_open_sync_with_config(
             return 3;
         }
     };
+    let uri: http::Uri = match primary_url.try_into(){
+        Ok(uri) => uri,
+        Err(e) => {
+            set_err_msg(format!("Wrong primary URL: {e}"), out_err_msg);
+            return 100;
+        }
```

Comment:
This is C bindings specific logic. core crate does not need it. It already has different builders for different versions of database.

Reply:
if I understand this code correctly, its converting some String to a Uri which we already do this logic for other config types https://docs.rs/libsql/latest/libsql/struct.Builder.html#method.new_remote as you can see here we accept a string. We should align them so that the bindings crate doesn't need to import http. I think that doesn't make much sense a consumer of `libsql` shouldn't have to import `http`, this may have been my mistake when I wrote the config.

Reply:
Ah yes, you're right I misread the code, I just went and checked the call site I was worried about. This is fine then thanks for the explanation.